### PR TITLE
Ron Swanson Spice IA

### PR DESCRIPTION
**Pull request by talsraviv**
**What does your Instant Answer do?**
This Spice provides random Ron Swanson quotes when his name is entered at the beginning or end of a query.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Ron Swanson can only truly be understood through his quotes - which traditional organic links do not highlight directly enough.

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://github.com/jamesseanwright/ron-swanson-quotes

**Why did you choose this data source?**
Found it on programmable web and I believe it is the only Ron Swanson API.

**Are there any other alternative (better) data sources?**
Not digitally.

**What are some example queries that trigger this Instant Answer?**
Ron Swanson's hair
I love Ron Swanson

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Blindly patriotic American 45-year-old men. Or anyone else with a sense of humor.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No

**Which existing Instant Answers will this one supersede/overlap with?**
The Wikipedia entry.

**Are you having any problems? Do you need our help with anything?**
I may need to choose a better template - this was the simplest. If there's precedents for what works well for quotes I'd rather use that.

Also, instead of handlebars or a normalize function, I'm using a self-calling function in the add() function to format my template variables. 

I tried using a `normalize` function to no avail (@moollaza we'll talk about this and a few other questions next week) The self-calling function was the only way I could see to do this using the text template. I'm sure there's a better way that I'm missing.

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**
No

**Where did you hear about DuckDuckHack? (For first time contributors)**

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screen shot 2015-03-27 at 2 35 46 pm

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information
[X] Wrote test file and added to t/ directory
[X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 8

    - Mac OSX
        [X] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
```

### CODE REVIEW

**Comment by zekiel** [comment]
@talsraviv  [slow clap]

File: share/spice/ron_swanson/ron_swanson.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        // Render the response
+        Spice.add({
+            id: "ron_swanson",
+            name: "Ron Swanson Quote",
+            data: { 
+                title: function(){
+                    return '"' + api_result.quote + '" - Ron Swanson';
+                }()
+            },
```
Comment (MrChrisW) [inline share/spice/ron_swanson/ron_swanson.js]: You can use the normalise function like this:
https://gist.github.com/MrChrisW/c03efafa711d61ef67ff

**Comment by MrChrisW** [comment]
I think the triggers should be limited to just "ron swanson" and "ron swanson quotes" at this stage to avoid over triggering.

File: share/spice/ron_swanson/ron_swanson.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        // Validate the response (customize for your Spice)
+        if (!api_result || api_result.error) {
+            return Spice.failed('ron_swanson');
+        }
+
+        // Render the response
+        Spice.add({
+            id: "ron_swanson",
+            name: "Ron Swanson Quote",
```
Comment (MrChrisW) [inline share/spice/ron_swanson/ron_swanson.js]: Maybe this could be changed to just `Quotation` or `Quotations`

**Comment by MrChrisW** [comment]
The only standard format for quote IA's we have at the moment is the [brainy_quote](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/share/spice/brainy_quote/brainy_quote.js) IA. It currently uses a custom template. [Example](https://duckduckgo.com/?q=Neil+Armstrong+Quotes&ia=quotations)

**Comment by zekiel** [comment]
good idea @MrChrisW

**Comment by talsraviv** [comment]
Ah cool - future built-in template? 

@MrChrisW about this PR - safe to assume my contributions to anything functional are meant as exercises to help understand DDH better - completing this will be a very low priority for me (unless someone has a pressing need to see Ron Swanson better represented in SERPs, speak now @zekiel)

**Comment by MrChrisW** [comment]
@zekiel :+1: Cool chatting the other day, It's a shame the video wasn't so good :frowning: 

>Ah cool - future built-in template? 

It's probably a good idea to standardise a quote template, that'll be up to DDG staff :+1: 

There's some outstanding design changes to the brainy quote template at the moment, see #1628 (trying to add a second quote and fix spacing issues)

It may be best to wait for these to be completed and replicate or just use the text template, @moollaza opinions ?

@talsraviv  No problem, I understand! I see your main priorities seems to be in documentation :wink:

**Comment by zekiel** [comment]
just a note @MrChrisW  to say that Paul is looking into better video options iirc :\  It's been like this w/ every client we've tried tho

**Comment by jagtalon** [comment]
Alright well I'm moving the code to https://github.com/duckduckgo/zeroclickinfo-spice/tree/ideas/ron-swanson if anybody wants to finish it! :sunglasses:

**Pull request closed without merging**